mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -69,70 +152,216 @@ def subset_variables(ds, vlist):  # {{{
+        An example:
+            isel_values = {'nVertLevels': slice(0, 3),
+                           'nCells': cellIDs}
+
+    year_offset : float
+        An offset used to convert an MPAS date to a date in the range supported
+        by xarray (numpy.datetime64).  Resulting dates must be between 1678
+        and 2622.
+
+    Returns
+    -------
+    ds : xarray.DataSet object
+        A copy of the data set with the time coordinate set and which
+        has been sliced.
+
+    Authors
+    -------
+    Phillip J. Wolfram, Milena Veneziani, Luke van Roekel
+    and Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/15/2017
+    """
+
+    datetimes = _get_datetimes(ds, time_variable_name,
+                               year_offset)
+
+    _assert_valid_datetimes(datetimes, year_offset)
+
+    # append the corret time information
+    ds.coords['Time'] = datetimes
+    # record the yroffset
+    ds.attrs.__setitem__('time_yearoffset', str(year_offset))
+
+    if variable_list is not None:
+        ds = subset_variables(ds,
+                              _ensure_list(variable_list))
+
+    _assert_valid_selections(ds, sel_values,
+                             isel_values)
+
+    if sel_values is not None:
+        ds = ds.sel(**sel_values)
+
+    if isel_values is not None:
+        ds = ds.isel(**isel_values)
 
     return ds  # }}}
 
 
-def assert_valid_datetimes(datetimes, yearoffset):  # {{{
+def remove_repeated_time_index(ds):  # {{{
+    """
+    Remove repeated times from xarray dataset.
+
+    Parameters
+    ----------
+    ds : xarray.DataSet object
+        The data set potentially containing repeated time indices.
+
+    Returns
+    -------
+    ds : xarray.DataSet object
+        A copy of the original data set with any repeated time indices removed.
+
+    Authors
+    -------
+    Phillip J. Wolfram, Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/10/2017
+    """
+    # get repeated indices
+    times = ds.Time.values
+    indices = range(len(times))
+    unique_times = set()
+    remove = []
+    for timeIndex, time in enumerate(times):
+        if time not in unique_times:
+            unique_times.add(time)
+        else:
+            remove.append(timeIndex)
+
+    # remove repeaded indices, working backwards from the last
+    remove.reverse()
+    for timeIndex in remove:
+        indices.pop(timeIndex)
+
+    # remove repeated indices
+    ds = ds.isel(Time=indices)
+
+    return ds  # }}}
+
+
+def _assert_valid_datetimes(datetimes, year_offset):  # {{{
     """
     Ensure that datatimes are compatable with xarray
 
-    Phillip J. Wolfram
-    04/20/2016
+    Authors
+    -------
+    Phillip J. Wolfram, Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/10/2017
     """
     assert datetimes[0].year > 1678, \
-        'ERROR: yearoffset={}'.format(yearoffset) + \
+        'ERROR: year_offset={}'.format(year_offset) + \
```

Comment:
Same comment as before.  This change is not helpful, but is confusing and against PEP8 and standard conventions.